src/lib/dot-matrix/DotMatrix.tsx

```diff
@@ -0,0 +1,83 @@
+import React from "react";
+import { v4 } from 'uuid';
+import classes from './styles.module.scss';
+import { DotMatrixPropType, DataPointType } from "./types";
+import { useDotMatrix } from './custom-hooks/useDotMatrix';
+import { Elements } from "./constants";
+
+const DotMatrix = (props: DotMatrixPropType): JSX.Element => {
+  const {
+    title,
+    dataPoints,
+    dimensions = {
+      rows: 5,
+      columns: 12
+    },
+    styles = {}
+  } = props;
+
+  const [data, total, partialVal] = useDotMatrix(dataPoints, dimensions);
+  const getStyles = (element: Elements): object => {
+    const getElementStyle = styles[element];
+    if (getElementStyle) {
+      return getElementStyle();
+    }
+    return {};
+  };
+
+  const getNumberOfDots = (point: DataPointType): number =>  {
+    const { rows = 5, columns = 12 } = dimensions;
```

Comment:
Do need to specify the count here?

Reply:
If the user specifies just one of the dimension, then the other value need to specified by us. For that this is required

Reply:
rows and columns are non-nullable, right? So both should be present?

Reply:
we have given the rows and columns as optional value